common/setups/serialization.py

```diff
@@ -112,6 +124,72 @@ def _sis_hash(self):
         return sis_hash_helper(self.code_object)
 
 
+class PartialImport(Import):
+    """
+    Like Import, but for partial callables where certain parameters are given fixed and are hashed.
+    """
+
+    TEMPLATE = textwrap.dedent(
+        """\
+            import functools
+            kwargs = ${KWARGS}
+    
+            from ${IMPORT_PATH} import ${IMPORT_NAME} as _${IMPORT_NAME}
+            ${OBJECT_NAME} = functools.partial(_${IMPORT_NAME}, **kwargs)
+    
+        """
```

Comment:
There are a lot of problems with this code. It exposes and leaks many symbols to the global namespace (`functools`, `kwargs`, `_${IMPORT_NAME}`). This could potentially break other code (admittedly unlikely, but still).

I just want to raise this. There are certainly easy ways to reduce this (without completely avoiding it) and there are also ways to completely avoid this (but making the code quite ugly). For example, you can use some combinations of `__import__` calls.

Reply:
This should be possible via `__import__` (or `importlib` if you like that more).

Reply:
What about:
```python
"""
            import functools
            ${OBJECT_NAME} = functools.partial(
                __import__("${IMPORT_PATH}", globals(), locals(), ["${IMPORT_NAME}"], 0).${IMPORT_NAME},
                **${KWARGS}
            )
"""
```
?

Reply:
`functools` itself can also be done this way.
Also `globals`, `locals` and `level` don't need to be passed.
Also `fromlist` better via kwarg.
So:
```python
${OBJECT_NAME} = __import__("functools").partial(
    __import__("${IMPORT_PATH}", fromlist=["${IMPORT_NAME}"]).${IMPORT_NAME},
    **${KWARGS}
)
```